bin/i18n/utils/sync_up_base.rb

```diff
@@ -0,0 +1,122 @@
+require 'optparse'
+
+require_relative '../i18n_script_utils'
+require_relative '../metrics'
+require_relative '../utils/crowdin_client'
+
+module I18n
+  module Utils
+    class SyncUpBase
+      Config = Struct.new :crowdin_project, :source_paths, :ignore_paths, keyword_init: true
+
+      MAIN_CROWDIN_PROJECT = 'codeorg'.freeze
+
+      def self.config
+        @config ||= Config.new(
+          crowdin_project: MAIN_CROWDIN_PROJECT,
+          source_paths: [],
+          ignore_paths: [],
+        )
+      end
+
+      def self.parse_options
+        options = Options.new
+
+        OptionParser.new do |opts|
+          opts.on('-t', '--testing', 'Run in testing mode') do
+            options[:testing] = true
+          end
+        end.parse!
+
+        options.to_h
+      end
```

Comment:
Assuming we will have similar options to parse in the sync out, could we have this `parse_options` method in `i18n_script_utils` so it can be shared between sync scripts?

Reply:
Yes, this is the idea. But for now, it's better to keep it there until the sync-out implementation